stock_quant_manual_assign/wizard/assign_manual_quants.py

```diff
@@ -0,0 +1,149 @@
+# Copyright 2015 Mikel Arregi - AvanzOSC
+# Copyright 2015 Oihane Crucelaegui - AvanzOSC
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from odoo import _, api, fields, models
+from odoo.addons import decimal_precision as dp
+from odoo.exceptions import UserError
+from odoo.tools.float_utils import float_compare
+
+
+class AssignManualQuants(models.TransientModel):
+    _name = 'assign.manual.quants'
+
+    @api.multi
+    @api.constrains('quants_lines')
+    def check_qty(self):
+        precision_digits = self.env[
+            'decimal.precision'].precision_get('Product Unit of Measure')
+        move = self.env['stock.move'].browse(self.env.context['active_id'])
+        for record in self.filtered('quants_lines'):
+            if float_compare(record.lines_qty, move.product_qty,
+                             precision_digits=precision_digits) > 0:
+                raise UserError(
+                    _('Quantity is higher than the needed one'))
+
+    @api.depends('quants_lines', 'quants_lines.qty')
+    def _compute_qties(self):
+        move = self.env['stock.move'].browse(self.env.context['active_id'])
+
+        lines_qty = sum(quant_line.qty for quant_line in self.quants_lines
+                        if quant_line.selected)
+        self.lines_qty = lines_qty
+        self.move_qty = move.product_qty - lines_qty
+
+    lines_qty = fields.Float(
+        string='Reserved qty', compute='_compute_qties',
+        digits=dp.get_precision('Product Unit of Measure'))
+    move_qty = fields.Float(string='Remaining qty', compute='_compute_qties',
+                            digits=dp.get_precision('Product Unit of Measure'))
+    quants_lines = fields.One2many('assign.manual.quants.lines',
+                                   'assign_wizard', string='Quants')
+
+    @api.multi
+    def assign_quants(self):
+        quant = self.env['stock.quant']
+        move = self.env['stock.move'].browse(self.env.context['active_id'])
```

Comment:
Try to make an attribute in the model that will be filled in with active_id (with a default_get). It will be accessible everywhere